app/validators/routes_and_redirects_validator.rb

```diff
@@ -147,11 +147,48 @@ def validate(location, redirect)
 
   private
 
+    def acceptable_destination?(target)
+      target.starts_with?("/") || valid_govuk_campaign_url?(target)
+    end
+
+    def valid_govuk_campaign_url?(target)
+      uri = URI.parse(target)
+      host = uri.host
+      if host =~ /\A.+\.campaign\.gov\.uk\z/i &&
+          %w(http https).include?(uri.scheme)
```

Comment:
Surely the campaigns platform will use HTTPS? We shouldn't be sending users to anything over unsecured HTTP if we can possibly avoid it.